### PR TITLE
Fix alternative subject identifier issues

### PR DESCRIPTION
**Pull request by sandushi**
### Purpose
- Fix the issues with alternative subject identifier

### Related Issues
-  https://github.com/wso2/product-is/issues/20138

### Related PRs
- 

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 1]
```diff
@@ -180,6 +182,22 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
         }
     }, [ selectedSubjectValue ]);
 
+    const getSelectedDropDownValue = ((
```
Comment (Achintha444) [inline features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:185]: add comment

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 2]
```diff
@@ -298,6 +316,11 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
         setMandateLinkedLocalAccount(value);
     };
 
+    const disableAlternativeSubjectIdentifier = (value: boolean) => {
```
Comment (Achintha444) [inline features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:319]: add comment

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 3]
```diff
@@ -422,17 +446,29 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
...
                         </>
                     ) }
+                    { !resolveSubjectAttributeHiddenStatus () && isEmpty(dropDownOptions) && (
+                        <Message info>
+                            <Icon name="info circle" />
+                            { t("applications:forms.advancedAttributeSettings" +
+                        ".sections.subject.fields.subjectAttribute.info") }
+                        </Message>
+                    ) }
+
```
Comment (Achintha444) [inline features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:456]:
```suggestion
```

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 4]
```diff
@@ -422,17 +446,29 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
...
                         }
+                        placeholder = {
+                            t("applications:forms.advancedAttributeSettings" +
+                                ".sections.subject.fields.subjectAttribute.placeholder")
+                        }
                         required={ claimMappingOn }
                         value={ selectedSubjectValue }
                         options={ dropDownOptions }
-                        hidden={ resolveSubjectAttributeHiddenStatus() }
+                        hidden={ resolveSubjectAttributeHiddenStatus() || isEmpty(dropDownOptions) }
```
Comment (Achintha444) [inline features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:471]: shall we get this to a separate function for readability ?

Comment (Achintha444) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:471]: check other places as well

File: features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx [hunk 5]
```diff
@@ -511,6 +511,21 @@ export const AttributeSelectionOIDC: FunctionComponent<AttributeSelectionOIDCPro
         );
     };
 
+    const isSelectedSubjectAttributeIncluded = (scope: OIDCScopesClaimsListInterface) : boolean => {
+
```
Comment (Achintha444) [inline features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx:515]:
```suggestion
```

File: features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx [hunk 6]
```diff
@@ -511,6 +511,21 @@ export const AttributeSelectionOIDC: FunctionComponent<AttributeSelectionOIDCPro
...
+            && selectedSubjectValue !== defaultSubjectAttribute;
+    };
+
+    const getSelectedSubjectAttributeClaimURI = (scope: OIDCScopesClaimsListInterface) : string => {
+
+        const selectedAttributeURI: string = scope.claims.find((claim:ExternalClaim ) =>
+            claim.mappedLocalClaimURI == selectedSubjectValue).localClaimDisplayName;
+
+        return selectedAttributeURI.charAt(0).toUpperCase() + selectedAttributeURI.slice(1);
+
```
Comment (Achintha444) [inline features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx:526]:
```suggestion
```

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 1 above)
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:188]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e73261ed2dbc028fa40283f153f2e178df3b5e85

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 2 above)
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx:341]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e73261ed2dbc028fa40283f153f2e178df3b5e85

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 3 above)
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e73261ed2dbc028fa40283f153f2e178df3b5e85

File: features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx (hunk 5 above)
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e73261ed2dbc028fa40283f153f2e178df3b5e85

File: features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx (hunk 6 above)
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e73261ed2dbc028fa40283f153f2e178df3b5e85

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 4 above)
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e73261ed2dbc028fa40283f153f2e178df3b5e85

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 7]
```diff
@@ -180,6 +182,22 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
...
+        const dropDownOptions: DropdownOptionsInterface[] = options as DropdownOptionsInterface[];
+        let claimURI: string = "";
+
+        if (showSubjectAttribute) {
+            claimURI = getDefaultDropDownValue (dropDownOptions, checkValue);
+        } else {
+            claimURI = defaultSubjectAttribute;
+        }
+
+        return claimURI;
```
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx]: Resolved with https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/attribute-list-item.tsx [hunk 8]
```diff
@@ -295,7 +301,7 @@ export const AttributeListItem: FunctionComponent<AttributeListItemPropInterface
                     >
                         <Checkbox
                             checked={ initialRequested || requested || subject }
-                            onClick={ !readOnly && handleRequestedCheckChange }
+                            onClick={ (!readOnly && !subject) ? handleRequestedCheckChange : ()=> null }
```
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-list-item.tsx]: Resolved with https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx [hunk 9]
```diff
@@ -422,17 +446,29 @@ export const AdvanceAttributeSettings: FunctionComponent<AdvanceAttributeSetting
                             </Hint>
                         </>
                     ) }
+                    { !resolveSubjectAttributeHiddenStatus () && isEmpty(dropDownOptions) && (
+                        <Message info>
+                            <Icon name="info circle" />
+                            { t("applications:forms.advancedAttributeSettings" +
+                        ".sections.subject.fields.subjectAttribute.info") }
```
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx [hunk 10]
```diff
@@ -558,7 +574,23 @@ export const AttributeSelectionOIDC: FunctionComponent<AttributeSelectionOIDCPro
                             </Header.Content>
                         )
                         : (
-                            <Header.Content>
+                            <Header.Content
+                                className= { isSelectedSubjectAttributeIncluded(scope)
```
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-selection-oidc.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/attribute-settings.tsx [hunk 11]
```diff
@@ -720,6 +720,27 @@ export const AttributeSettings: FunctionComponent<AttributeSettingsPropsInterfac
         }
     };
 
+    /**
+     * Create sorted dropdown list.
+     */
+    const createDropDownList = () : DropdownOptionsInterface[]=> {
```
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-settings.tsx]: resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/advance-attribute-settings.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

File: features/admin.applications.v1/components/settings/attribute-management/attribute-settings.tsx [hunk 12]
```diff
@@ -825,24 +846,28 @@ export const AttributeSettings: FunctionComponent<AttributeSettingsPropsInterfac
...
-                            displayName={ element.localClaimDisplayName }
-                            claimURI={ element.claimURI }
-                            value={ element.mappedLocalClaimURI }
-                        />
-                    ),
-                    value: element.mappedLocalClaimURI
-                };
+            unfilteredExternalClaimsGroupedByScopes.map((scope: OIDCScopesClaimsListInterface) => {
+                scope?.claims.map((element: ExtendedExternalClaimInterface) => {
+                    if(element.requested) {
```
[the remark this answers is not in the document]
Comment (sandushi) [reply features/admin.applications.v1/components/settings/attribute-management/attribute-settings.tsx]: Resolved by https://github.com/wso2/identity-apps/pull/5879/commits/e96765a86f2734daed636d4a9962d98a0b703bd5

**Review by DonOmalVindula — APPROVED** [review]

**Pull request merged**